pkg/controller/endpointslice/reconciler_test.go

```diff
@@ -460,34 +459,152 @@ func TestReconcile1Pod(t *testing.T) {
+			wantUpdate:  true,
+			wantMetrics: expectedMetrics{desiredSlices: 1, actualSlices: 1, desiredEndpoints: 0, addedPerSync: 0, removedPerSync: 0, numCreated: 1, numUpdated: 0, numDeleted: 0, slicesChangedPerSync: 1},
+		},
+		{
+			desc:        "Existing placeholder that's the same",
+			existing:    newEndpointSlice(&svc, &endpointMeta{Ports: []discovery.EndpointPort{}, AddressType: discovery.AddressTypeIPv4}),
+			wantMetrics: expectedMetrics{desiredSlices: 0, actualSlices: 0, desiredEndpoints: 0, addedPerSync: 0, removedPerSync: 0, numCreated: 0, numUpdated: 0, numDeleted: 0, slicesChangedPerSync: 0},
+		},
+		{
+			desc:        "Existing placeholder that's different",
+			existing:    newEmptyEndpointSlice(1, namespace, epMeta, svc),
+			wantUpdate:  true,
+			wantMetrics: expectedMetrics{desiredSlices: 1, actualSlices: 1, desiredEndpoints: 0, addedPerSync: 0, removedPerSync: 0, numCreated: 0, numUpdated: 1, numDeleted: 0, slicesChangedPerSync: 1},
+		},
+	}
 
-	numActionsBefore := len(client.Actions())
-	r := newReconciler(client, []*corev1.Node{{ObjectMeta: metav1.ObjectMeta{Name: "node-1"}}}, defaultMaxEndpointsPerSlice)
-	reconcileHelper(t, r, &svc, []*corev1.Pod{}, []*discovery.EndpointSlice{endpointSlice1}, time.Now())
-	assert.Len(t, client.Actions(), numActionsBefore+1, "Expected 1 additional clientset action")
-	actions := client.Actions()
-	assert.True(t, actions[numActionsBefore].Matches("update", "endpointslices"), "Action should be update endpoint slice")
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			client := newClientset()
+			setupMetrics()
 
-	slices := fetchEndpointSlices(t, client, namespace)
-	assert.Len(t, slices, 1, "Expected 1 endpoint slices")
+			existingSlices := []*discovery.EndpointSlice{}
+			if tc.existing != nil {
+				existingSlices = append(existingSlices, tc.existing)
+				_, createErr := client.DiscoveryV1().EndpointSlices(namespace).Create(context.TODO(), tc.existing, metav1.CreateOptions{})
+				assert.Nil(t, createErr, "Expected no error creating endpoint slice")
+			}
 
-	assert.Regexp(t, "^"+svc.Name, slices[0].Name)
-	assert.Equal(t, svc.Name, slices[0].Labels[discovery.LabelServiceName])
-	assert.EqualValues(t, []discovery.EndpointPort{}, slices[0].Ports)
-	assert.EqualValues(t, []discovery.Endpoint{}, slices[0].Endpoints)
-	expectTrackedGeneration(t, r.endpointSliceTracker, &slices[0], 1)
-	expectMetrics(t, expectedMetrics{desiredSlices: 1, actualSlices: 1, desiredEndpoints: 0, addedPerSync: 0, removedPerSync: 0, numCreated: 0, numUpdated: 1, numDeleted: 0, slicesChangedPerSync: 1})
+			numActionsBefore := len(client.Actions())
+			r := newReconciler(client, []*corev1.Node{{ObjectMeta: metav1.ObjectMeta{Name: "node-1"}}}, defaultMaxEndpointsPerSlice)
+
+			// Run reconcile multiple times to make sure that subsequent calls don't do any additional work
+			for i := 0; i < 1; i++ {
+				reconcileHelper(t, r, &svc, []*corev1.Pod{}, existingSlices, time.Now())
+			}
```

Comment:
:upside_down_face: multiple times?  this only executes once  ;)

Reply:
Whoops forgot to remove this from an earlier version of the test :)

Reply:
Done